kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableImpl.java

```diff
@@ -237,6 +239,13 @@ public HasMetadata waitUntilReady(long amount, TimeUnit timeUnit) throws Interru
     return h.waitUntilReady(client, config, meta.getMetadata().getNamespace(), meta, amount, timeUnit);
   }
 
+  @Override
+  public HasMetadata waitUntilCondition(Predicate<HasMetadata> condition, long amount,
+    TimeUnit timeUnit) throws InterruptedException {
+    //TODO what should be done here?
+    throw new NotImplementedException();
```

Comment:
Thanks for the info! I'll look into it soon

Reply:
Thank you for the suggestion!

I have another question however:

`NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableListImpl` seems to be doing some sophisticated things for waiting. Any idea what should be done there?

Reply:
And another, won't changes to `ResourceHandler` mean that the `addon` clients like Istio and ServiceCatalog break?
Just want to make sure, breaking them probably isn't a problem since we can fix them easily

Reply:
I have updated `ResourceHandler` and all that remains now is to implement waitForCondition for `NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableListImpl`. 
@iocanel any pointers?

Reply:
Yes, extensions will break that's inevitable. As long as the change is reflected in the version  (we need to bump the major version),  we are fine.

Reply:
Regarding `NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableListImpl` I would just follow the logic of `waitUntilReady`.

Reply:
Thanks, hopefully I'll pick it up again soon

Reply:
Looking at the code more closely, I know understand how it works :).

However, we have a different problem... 
`NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableListImpl` implements `Waitable<List<HasMetadata>>` which means that it needs to accept Predicate<List<HasMetadata>>.

However what we need to pass to ResourceHandler is  `Waitable<HasMetadata>`...
I would not like to try to circumvent generics to achieve what we want.

@iocanel Any suggestions on how to proceed?